.github/workflows/build-ros2-ws-action.yml

```
@@ -0,0 +1,57 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [master]  
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        run: git clone -b feature/fix-using-grpc https://github.com/ichiro-its/aruku.git
+
+      - name: Checkout Kansei
+        run: git clone https://github.com/ichiro-its/kansei.git                 
+
+      - name: Checkout Kansei Interfaces
+        run: git clone https://github.com/ichiro-its/kansei_interfaces.git         
+      
+      - name: Checkout Aruku Interfaces
+        run: git clone https://github.com/ichiro-its/aruku_interfaces.git     
+
+      - name: Checkout Keisan
+        run: git clone https://github.com/ichiro-its/keisan.git        
+
+      - name: Checkout tachimawari
+        run: git clone https://github.com/ichiro-its/tachimawari.git  
+
+      - name: Checkout tachimawari interfaces
+        run: git clone https://github.com/ichiro-its/tachimawari_interfaces.git    
```

Since you proposed using the `git clone` command, as in my previous comment, I suggest running all commands in the same step:

```suggestion
      - name: Checkout dependencies
        run: |
          git clone https://github.com/ichiro-its/kansei.git                 
          git clone https://github.com/ichiro-its/kansei_interfaces.git
          git clone https://github.com/ichiro-its/aruku_interfaces.git
          git clone https://github.com/ichiro-its/keisan.git
          git clone https://github.com/ichiro-its/tachimawari.git
          git clone https://github.com/ichiro-its/tachimawari_interfaces.git
```